server/src/main/java/org/apache/lucene/index/BaseSingleStarTreeBuilder.java

```diff
@@ -0,0 +1,660 @@
+            try {
+                dimensionReaders[i].nextDoc();
+            } catch (IOException e) {
+                logger.error("unable to iterate to next doc", e);
+            }
+
+            if (starTreeField.getDimensionsOrder().get(i) instanceof DateDimension) {
+                dimensions[i] = handleDateDimension(
+                    dimensionsSplitOrder[i],
+                    starTreeDocValuesIteratorFactory.getNextValue(dimensionReaders[i])
+                );
+            } else {
+                dimensions[i] = starTreeDocValuesIteratorFactory.getNextValue(dimensionReaders[i]);
+            }
+        }
+        return dimensions;
+    }
+
+    /**
+     * Returns the next segment star-tree document for the metrics
+     *
+     * @return metric values for each of the star-tree metric
+     * @throws IOException when we are unable to iterate to the next doc
+     */
+    private Object[] getNextSegmentStarTreeDocumentMetrics() throws IOException {
+        Object[] metrics = new Object[numMetrics];
+        for (int i = 0; i < numMetrics; i++) {
+            // Ignore the column for COUNT aggregation function
+            if (metricReaders[i] != null) {
+                try {
+                    metricReaders[i].nextDoc();
+                } catch (IOException e) {
+                    // TODO : handle null values in columns
+                    logger.error("unable to iterate to next doc", e);
+                }
+                metrics[i] = starTreeDocValuesIteratorFactory.getNextValue(metricReaders[i]);
+            }
+        }
+        return metrics;
+    }
+
+    /**
+     * Merges a segment star-tree document (raw) into the aggregated star-tree document.
+     *
+     * <p>Will create a new aggregated star-tree document if the current one is {@code null}.
+     *
+     * @param aggregatedStarTreeDocument Aggregated star-tree document
+     * @param segmentStarTreeDocument    Segment star-tree document
+     * @return Merged starTreeDocument
+     */
+    protected StarTreeDocument aggregateSegmentStarTreeDocument(
+        StarTreeDocument aggregatedStarTreeDocument,
+        StarTreeDocument segmentStarTreeDocument
+    ) {
+        // TODO: HANDLE KEYWORDS LATER!
+        if (aggregatedStarTreeDocument == null) {
+            long[] dimensions = Arrays.copyOf(segmentStarTreeDocument.dimensions, numDimensions);
+            Object[] metrics = new Object[numMetrics];
+            for (int i = 0; i < numMetrics; i++) {
+                try {
+                    StarTreeNumericType numericType = StarTreeNumericType.fromNumericType(numericTypes[i]);
+                    metrics[i] = valueAggregators[i].getInitialAggregatedValue((Long) segmentStarTreeDocument.metrics[i], numericType);
+                } catch (IllegalArgumentException | NullPointerException e) {
+                    logger.error("Cannot parse initial aggregated value", e);
+                    throw new IllegalArgumentException(
```

Comment:
At this point, we are in flush phase of indexing. I don't think "IllegalArgumentException" is the right exception we need to throw. Lets check the standard practice in such cases.

Reply:
Illegal argument seemed fit since the nature of value is question is not permissible like [this](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/index/SortedNumericDocValuesWriter.java#L211).
Will check if there is a more suitable exception than IllegalArgument / Runtime Exception

Reply:
Replaced with IllegalStateException